src/Storages/MergeTree/IMergedBlockOutputStream.cpp

```diff
@@ -79,10 +79,17 @@ NameSet IMergedBlockOutputStream::removeEmptyColumnsFromPart(
     }
 
     /// Remove files on disk and checksums
-    for (const String & removed_file : remove_files)
+    for (auto itr = remove_files.begin(); itr != remove_files.end();)
     {
-        if (checksums.files.contains(removed_file))
-            checksums.files.erase(removed_file);
+        if (checksums.files.contains(*itr))
+        {
+            checksums.files.erase(*itr);
```

Comment:
```suggestion
            checksums.files.erase(*itr);
            ++itr;
```
?

Reply:
Thank you, changed this code 10 times to reproduce the bug with test and finally forget normal case :)